modules/core/src/smithy4s/internals/DocumentDecoderSchemaVisitor.scala

```diff
@@ -307,66 +308,120 @@ class DocumentDecoderSchemaVisitor(
     }
   }
 
+  trait UnknownFieldsDecoder[A] { self =>
```

Comment:
This one could be removed too : just compile the document decoder for `A` and pass it a `DObject` containing the `unknownFields`

Reply:
Done in https://github.com/disneystreaming/smithy4s/commit/59e0d225da28823f1679bc8b7243e0deab05a75a.